internal/neofs/neofs.go

```diff
@@ -46,16 +47,23 @@ type NeoFS struct {
 	anonSigner  user.Signer
 	cfg         Config
 	epochGetter EpochGetter
+	buffers     chan []byte
 }
 
 // NewNeoFS creates new NeoFS using provided pool.Pool.
 func NewNeoFS(p *pool.Pool, signer user.Signer, anonSigner user.Signer, cfg Config, epochGetter EpochGetter) *NeoFS {
+	buffers := make(chan []byte, cfg.UploadChunkPoolLength)
```

Comment:
https://pkg.go.dev/sync#Pool ?

Reply:
[This option](https://github.com/nspcc-dev/neofs-s3-gw/pull/835/commits/ce7d2d84be14c655882371851085b27e03447117) better then original by memory, but lose by CPU.
I have to adopt it, in the code, it looks better, but the numbers are not so impressive.

The first variant with the channels has one possible bottleneck - reading from the channel. Concurrence is limited by channel capacity. We may try to allocate a new slice if the channel is locked. Unfortunately, it is a new allocation...

The second option is trying to experiment with channel sizes with different bench loads.

Reply:
After some benchmarks, `sync.Pool` won the `channel` version. Updated